node-commons/src/main/java/org/apache/iotdb/commons/partition/DataPartition.java

```diff
@@ -238,11 +248,73 @@ public void createDataPartition(
         .put(timePartitionSlot, Collections.singletonList(regionReplicaSet));
   }
 
-  public void serialize(ByteBuffer buffer) {
-    // TODO: Serialize DataPartition
+  public void serialize(ByteBuffer buffer) throws IOException, TException {
+    try (ByteArrayOutputStream byteArrayOutputStream = new ByteArrayOutputStream()) {
+      for (Entry<
+              String, Map<TSeriesPartitionSlot, Map<TTimePartitionSlot, List<TRegionReplicaSet>>>>
+          entry : dataPartitionMap.entrySet()) {
+        ReadWriteIOUtils.write(entry.getKey(), byteArrayOutputStream);
+        try (TIOStreamTransport tioStreamTransport =
+            new TIOStreamTransport(byteArrayOutputStream)) {
+          TProtocol protocol = new TBinaryProtocol(tioStreamTransport);
+          for (Entry<TSeriesPartitionSlot, Map<TTimePartitionSlot, List<TRegionReplicaSet>>>
+              entry1 : entry.getValue().entrySet()) {
+            entry1.getKey().write(protocol);
+            for (Entry<TTimePartitionSlot, List<TRegionReplicaSet>> entry2 :
+                entry1.getValue().entrySet()) {
+              entry2.getKey().write(protocol);
+              ReadWriteIOUtils.write(entry2.getValue().size(), byteArrayOutputStream);
+              entry2
+                  .getValue()
+                  .forEach(
+                      x -> {
+                        try {
+                          x.write(protocol);
+                        } catch (TException e) {
+                          throw new RuntimeException(e);
+                        }
+                      });
+            }
+          }
+        }
+      }
+      byte[] toArray = byteArrayOutputStream.toByteArray();
+      buffer.putInt(toArray.length);
+      buffer.put(toArray);
+    }
   }
 
-  public void deserialize(ByteBuffer buffer) {
-    // TODO: Deserialize DataPartition
+  public void deserialize(ByteBuffer buffer) throws TException, IOException {
```

Comment:
Don't forget to add tests for deserialize~

Reply:
fixed